content/chapters/compute/lecture/slides.md

```diff
@@ -0,0 +1,1161 @@
+* Efficient scheduling
+* Encapsulation and separation of tasks:
+    * VAS: isolate data
+    * CPU: computations, threads
+    * I/O: network packets, files
+
+----
+
+### What We Don't Want
+
+* Unresponsive processes
+* Processes monopolizing resources
+* System freezes
+
+---
+
+### Process Address Space
+
+* Each process has its own VAS:
+    * Isolation
+    * The process "thinks" it uses the entire address space
+
+----
+
+### Process Address Space
+
+![Process VAS](media/vas.svg)
+
+---
+
+## Process Attributes - from User Space
+
+* PID: numeric identifier
+* PPID: parent's PID (one parent per process)
+* Running time
+* State: we will discuss this later
+
+```
+student@os:~/.../compute/lecture/demo/create-thread$ python3 create_process.py
+[main] PID = 16066; PPID = 7820
+[thread] PID = 16066; PPID = 7820
+```
+
+----
+
+### Process Attributes - from User Space
+
+`ps`:
+```
+student@os:~$ ps -e --forest -o pid,ppid,rss,vsz,etime,stat,comm
+  PID  PPID   RSS    VSZ     ELAPSED STAT COMMAND
+    1     0   600    988  2-02:44:02 Sl   init
+  914     1    88    900  2-02:14:06 S    init
+ 2582   914  3376  82956  1-04:15:56 Ss    \_ gpg-agent
+19812     1   176    988    09:03:34 Ss   init
+19813 19812   176    988    09:03:34 S     \_ init
+19814 19813  6420  11256    09:03:34 Ss        \_ bash
+21713 19814  3132  10696       00:00 R+            \_ ps
+20650     1   176    988    05:43:41 Ss   init
+20651 20650   176    988    05:43:41 S     \_ init
+20652 20651   604   2612    05:43:41 Ss+       \_ sh
+20653 20652   536   2612    05:43:41 S+            \_ sh
+20658 20653   604   2612    05:43:41 S+                \_ sh
+20661 20658 74072 931768    05:43:41 Sl+                   \_ node
+20681 20661 44316 614424    05:43:40 Sl+                       \_ node
+20722 20661 39804 828152    05:43:40 Sl+                       \_ node
+20733 20661 147004 845832   05:43:40 Sl+                       \_ node
+20796 20733 119336 959952   05:43:38 Sl+                           \_ node
+21149 20733 41080 581504    04:31:20 Sl+                           \_ node
+20672     1   176    988    05:43:40 Ss   init
+20673 20672   176    988    05:43:40 S     \_ init
+20674 20673 41556 585172    05:43:40 Ssl+      \_ node
+20701     1   176    988    05:43:40 Ss   init
+20702 20701   176    988    05:43:40 S     \_ init
+20703 20702 48060 590308    05:43:40 Ssl+      \_ node
+20915     1   176    988    05:24:46 Ss   init
+20916 20915   176    988    05:24:46 S     \_ init
+20917 20916  5268  10188    05:24:46 Ss        \_ bash
+20964 20917 19840 516296    05:24:32 S+            \_ python3
+20981     1   176    988    05:22:54 Ss   init
+20982 20981   176    988    05:22:54 S     \_ init
+20983 20982  5100  10056    05:22:54 Ss+       \_ bash
+```
+
+----
+
+### Process Attributes - from Kernel Space
+
+* PCB (Process Control Block):
+    * Data structure in the kernel - `task_struct`
+    * All information about the process (from `ps`, `/proc`):
```

Comment:
Why is there a colon (`:`)?

Reply:
It was meant to be a hint to look at the next slide, but in reality, it looks stupid. I've removed it.